src/components/TextInput.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { ReactElement } from 'react';
+import {
+    FormControl as FormControlMUI,
+    InputLabel as InputLabelMUI,
+    Input as InputMUI,
+    FilledInput as FilledInputMUI,
+    SxProps,
+    Theme,
+} from '@mui/material';
+
+interface TextInputProps {
+    /**
+     * HTML element id
+     */
+    id?: string;
+    /**
+     * HTML input element type, defaults to text
+     */
+    type?: 'text' | 'username' | 'email' | 'password' | 'search' | 'url';
+    /**
+     * HTML element name
+     */
+    name?: string;
+    /**
+     * Label to display in the input field
+     */
+    label?: string;
+    /**
+     * HTML autocomplete attribute
+     */
+    autoComplete?: string;
+    /**
+     * Controls variant of MUI Input component, defaults to filled
+     */
+    variant?: 'filled' | 'outlined';
+    /**
+     * MUI color of button, defaults to `secondary`
+     */
+    color?: 'primary' | 'secondary' | 'success' | 'error' | 'info' | 'warning';
+    /**
+     * Value of controlled input
+     */
+    value?: string;
+    /**
+     * If true, the input will indicate an error
+     */
+    error?: boolean;
+    /**
+     * If true, the input element is required
+     */
+    required?: boolean;
+    /**
+     * Function to be run when input value changes
+     */
+    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    /**
+     * Function to set email error to false when input focused
+     */
+    onFocus?: () => void;
+    /**
+     * Icon to show in the input field
+     */
+    endAdornment?: ReactElement;
+    /**
+     * Provides an accessible label to the element
+     */
+    'aria-label'?: string;
```

Comment:
I don't think I have seen a prop structured this this. Why not `ariaLabel?: string;`?

Reply:
This was an assumption on my part. The aria-label prop is passed to the component exactly like `aria-label` so I assumed that's how it should look in the props interface. I'm not quite sure how to phrase it here but let me try something and let me know what you think.